src/keria/app/agenting.py

```diff
@@ -93,7 +95,7 @@ def setup(name, bran, adminPort, bootPort, base='', httpPort=None, configFile=No
         ending.loadEnds(agency=agency, app=happ)
         indirecting.loadEnds(agency=agency, app=happ)
 
-        server = http.Server(port=httpPort, app=happ)
+        server = createHttpServer(httpPort, happ, keypath, certpath, cafilepath)
```

Comment:
If you are going to change this to HTTPS, you will need to change the way you expose the OOBIs so that they are also HTTPs.  Otherwise others will not be able to resolve the oobis for this Agent.

This is brings to mind that we might need multiple cert options because users will probably want to expose this endpoint to the world on one domain name and expose the Admin API on another domain name.  Heck, we may need 3.

Reply:
Does Rodolfo's PR #91 address this need?

Reply:
No it does not.  You will have to update OobiResourceEnd.on_get to change what scheme (currently hardcoded to http) is used when generating OOBIs, based on the configuration of these flags.  Endpoints that are returned when OOBIs are resolved are controlled by configuration files so they are fine, but not OOBIs themselves.

Reply:
I just hit this problem. I'm getting a `socket.gaierror: [Errno -2] Name does not resolve` when resolving an HTTPS OOBI.
I'll get a PR submitted on this ASAP.

Reply:
I was able to use the changes from #91 for a fully HTTPS KERIA. I haven't yet tried switching my witness over to HTTPS. That's my next task.

Reply:
After switching my witness over to HTTPS I was able to get a KERIA HTTPS agent server and a KERIpy HTTPS witness server to work using the code in this PR as well as Rodolfo's changes from #91.

Reply:
You still have not address my change request.  That change needs to be made before this PR can be accepted.

Reply:
Do you want to merge #103 and have me re-base this on development or should I just pull those changes into this branch?

Reply:
Ok, I'll pull just the changes I need into this PR.

Reply:
Those changes are now committed to this PR.

Reply:
nevermind, tests are breaking. I'll fix it.